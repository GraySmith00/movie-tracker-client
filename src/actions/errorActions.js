export const setRegisterErrorState = errorMessage => ({
  type: 'SET_REGISTER_ERROR_STATE',
  errorMessage
});

export const setLoginErrorState = errorMessage => ({
  type: 'SET_LOGIN_ERROR_STATE',
  errorMessage
});

export const setFavoritesErrorState = errorMessage => ({
  type: 'SET_FAVORITES_ERROR_STATE',
  errorMessage
});
