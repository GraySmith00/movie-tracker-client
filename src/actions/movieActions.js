export const addNowPlaying = movies => ({
  type: 'ADD_NOW_PLAYING',
  movies
});

export const toggleMovieStatus = changedMovie => ({
  type: 'TOGGLE_MOVIE_STATUS',
  changedMovie
});

export const addFavoriteToState = movieId => ({
  type: 'ADD_FAVORITE_TO_STATE',
  movieId
});

export const removeFavoriteFromState = movieId => ({
  type: 'REMOVE_FAVORITE_FROM_STATE',
  movieId
});

export const populateFavoritesState = movieIds => ({
  type: 'POPULATE_FAVORITES_STATE',
  movieIds
});

export const clearFavorites = () => ({
  type: 'CLEAR_FAVORITES'
});
