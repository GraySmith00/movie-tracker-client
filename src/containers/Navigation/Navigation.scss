.brand {
  display: flex;
  justify-content: start;
  padding: 0;
  margin: 1rem 0 0 0;

  width: 100%;
  padding: 2rem;

  font-size: 1.5rem;

  text-decoration: none;
  color: #fff;
}

i {
  margin-right: 0.5rem;
}

.nav-btns {
  display: flex;
  flex-direction: column;

  margin-top: 10rem;
}

.nav-link {
  display: flex;
  justify-content: start;

  font-size: 1.2rem;
  text-decoration: none;
  color: #fff;
  width: 100%;
  padding: 2rem 0 2rem 2rem;
}

.active {
  background-color: rgba(0, 0, 0, 0.5);
}

.nav-link:hover {
  background-color: rgba(0, 0, 0, 0.5);
}

@media screen and (max-width: 780px) {
  .nav-btns {
    flex-direction: row;
    margin-top: 0;
    justify-content: space-evenly;
  }
  .nav-btn-wrapper {
    display: flex;
    flex-direction: row;
  }
  .brand {
    justify-content: center;
    font-size: 2rem;
    margin: 0 0 0 0;
  }
}
