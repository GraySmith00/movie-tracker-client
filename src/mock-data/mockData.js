const nowPlayingResultsArray = {
  results: [
    {
      vote_count: 507,
      id: 345940,
      video: false,
      vote_average: 6.2,
      title: "The Meg",
      popularity: 173.315,
      poster_path: "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg",
      original_language: "en",
      original_title: "The Meg",
      genre_ids: [28, 878, 53, 27],
      backdrop_path: "/ibKeXahq4JD63z6uWQphqoJLvNw.jpg",
      adult: false,
      overview:
        "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
      release_date: "2018-08-09"
    },
    {
      vote_count: 1095,
      id: 353081,
      video: false,
      vote_average: 7.4,
      title: "Mission: Impossible - Fallout",
      popularity: 138.478,
      poster_path: "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
      original_language: "en",
      original_title: "Mission: Impossible - Fallout",
      genre_ids: [12, 28, 53],
      backdrop_path: "/5qxePyMYDisLe8rJiBYX8HKEyv2.jpg",
      adult: false,
      overview:
        "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
      release_date: "2018-07-25"
    },
    {
      vote_count: 2833,
      id: 351286,
      video: false,
      vote_average: 6.6,
      title: "Jurassic World: Fallen Kingdom",
      popularity: 102.979,
      poster_path: "/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
      original_language: "en",
      original_title: "Jurassic World: Fallen Kingdom",
      genre_ids: [28, 12, 878],
      backdrop_path: "/3s9O5af2xWKWR5JzP2iJZpZeQQg.jpg",
      adult: false,
      overview:
        "Several years after the demise of Jurassic World, a volcanic eruption threatens the remaining dinosaurs on the island of Isla Nublar. Claire Dearing, the former park manager and founder of the Dinosaur Protection Group, recruits Owen Grady to help prevent the extinction of the dinosaurs once again.",
      release_date: "2018-06-06"
    }
  ]
};

export const modifiedNowPlaying = [
  {
    id: 345940,
    title: "The Meg",
    releaseDates: "2018-08-09",
    overview:
      "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
    poster_path: "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg"
  },
  {
    id: 353081,
    title: "Mission: Impossible - Fallout",
    img: "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
    overview:
      "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
    releaseDates: "2018-07-25"
  },
  {
    id: 351286,
    title: "Jurassic World: Fallen Kingdom",
    img: "http://image.tmdb.org/t/p/original/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
    realeaseDates: "2018-06-06",
    overview:
      "Several years after the demise of Jurassic World, â€¦event the extinction of the dinosaurs on the island of Isla Nublar. Claire Dearing, the former park manager and founder of the Dinosaur Protection Group, recruits Owen Grady to help prevent the extinction of the dinosaurs once again."
  }
];
