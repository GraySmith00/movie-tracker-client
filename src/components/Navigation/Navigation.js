import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, Link, withRouter } from 'react-router-dom';
import { setCurrentUser } from '../../actions/userActions';
import { clearFavorites } from '../../actions/movieActions';

import './Navigation.css';

export class Navigation extends Component {
  logoutUser = () => {
    const { setCurrentUser, clearFavorites, history } = this.props;
    setCurrentUser(null);
    clearFavorites();
    history.replace('/');
  };

  render() {
    return (
      <section className="navigation">
        <Link exact to="/" className="nav-link brand">
          <i className="fas fa-film" />
          MovieTracker
        </Link>
        <nav className="nav-btns">
          {this.props.currentUser !== null ? (
            <div>
              <NavLink exact to="/favorites" className="nav-link">
                <i className="fas fa-heart" />
                Favorites
              </NavLink>
              <Link
                exact
                to="/"
                className="nav-link nav-link-logout"
                onClick={this.logoutUser}
              >
                Logout
              </Link>
            </div>
          ) : (
            <div>
              <NavLink
                exact
                to="/register"
                className="nav-link"
                onClick={this.checkUserStatus}
              >
                <i className="fas fa-user-plus" />
                Sign Up
              </NavLink>
              <NavLink
                exact
                to="/login"
                className="nav-link"
                onClick={this.checkUserStatus}
              >
                <i className="fas fa-sign-in-alt" />
                Login
              </NavLink>
            </div>
          )}
        </nav>
      </section>
    );
  }
}

Navigation.propTypes = {
  setCurrentUser: PropTypes.func.isRequired,
  clearFavorites: PropTypes.func.isRequired,
  currentUser: PropTypes.object.isRequired
};

export const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user)),
  clearFavorites: () => dispatch(clearFavorites())
});

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Navigation)
);
