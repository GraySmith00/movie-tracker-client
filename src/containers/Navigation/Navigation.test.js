import React from 'react';
import { shallow } from 'enzyme';
import { Navigation, mapDispatchToProps } from './Navigation';
import { mockStore } from '../../mockData/mockStore';
import { clearFavorites } from '../../actions/movieActions';

describe('Navigation', () => {
  let wrapper;
  let mockSetCurrentUser;
  let mockClearFavorites;
  let mockHistory;

  beforeEach(() => {
    mockSetCurrentUser = jest.fn();
    mockClearFavorites = jest.fn();
    mockHistory = {
      length: 14,
      action: 'PUSH',
      location: { pathname: '/', search: '', hash: '', key: 'opw812' },
      replace: jest.fn()
    };

    wrapper = shallow(
      <Navigation
        setCurrentUser={mockSetCurrentUser}
        clearFavorites={mockClearFavorites}
        currentUser={null}
        history={mockHistory}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when currentUser exists', () => {
    wrapper = shallow(
      <Navigation
        setCurrentUser={mockSetCurrentUser}
        clearFavorites={mockClearFavorites}
        currentUser={mockStore.currentUser}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should setCurrentUser and clearFavorites when logOut user is called', () => {
    wrapper.instance().logoutUser();

    expect(mockSetCurrentUser).toHaveBeenCalledWith(null);
    expect(mockClearFavorites).toHaveBeenCalled();
  });

  it('should map to store on mapStateToDispatch of setCurrentUser', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.setCurrentUser(mockStore.currentUser);

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should map to store on mapStateToDispatch of setCurrentUser', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.clearFavorites();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
