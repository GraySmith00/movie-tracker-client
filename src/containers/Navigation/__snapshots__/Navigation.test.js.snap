// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    clearFavorites={[Function]}
    currentUser={null}
    history={
        Object {
            "action": "PUSH",
            "length": 14,
            "location": Object {
              "hash": "",
              "key": "opw812",
              "pathname": "/",
              "search": "",
            },
            "replace": [Function],
          }
    }
    setCurrentUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="nav-link brand"
          replace={false}
          to="/"
>
          <i
                    className="fas fa-film"
          />
          MovieTracker
</Link>,
        <nav
          className="nav-btns"
>
          <div>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav-link"
                              exact={true}
                              onClick={undefined}
                              to="/register"
                    >
                              <i
                                        className="fas fa-user-plus"
                              />
                              Sign Up
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav-link"
                              exact={true}
                              onClick={undefined}
                              to="/login"
                    >
                              <i
                                        className="fas fa-sign-in-alt"
                              />
                              Login
                    </NavLink>
          </div>
</nav>,
      ],
      "className": "navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-film"
/>,
            "MovieTracker",
          ],
          "className": "nav-link brand",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-film",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "MovieTracker",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        onClick={undefined}
                        to="/register"
            >
                        <i
                                    className="fas fa-user-plus"
                        />
                        Sign Up
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        onClick={undefined}
                        to="/login"
            >
                        <i
                                    className="fas fa-sign-in-alt"
                        />
                        Login
            </NavLink>
</div>,
          "className": "nav-btns",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                exact={true}
                onClick={undefined}
                to="/register"
>
                <i
                                className="fas fa-user-plus"
                />
                Sign Up
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                exact={true}
                onClick={undefined}
                to="/login"
>
                <i
                                className="fas fa-sign-in-alt"
                />
                Login
</NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": Array [
                  <i
                    className="fas fa-user-plus"
/>,
                  "Sign Up",
                ],
                "className": "nav-link",
                "exact": true,
                "onClick": undefined,
                "to": "/register",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-user-plus",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Sign Up",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": Array [
                  <i
                    className="fas fa-sign-in-alt"
/>,
                  "Login",
                ],
                "className": "nav-link",
                "exact": true,
                "onClick": undefined,
                "to": "/login",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-sign-in-alt",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Login",
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="nav-link brand"
            replace={false}
            to="/"
>
            <i
                        className="fas fa-film"
            />
            MovieTracker
</Link>,
          <nav
            className="nav-btns"
>
            <div>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="nav-link"
                                    exact={true}
                                    onClick={undefined}
                                    to="/register"
                        >
                                    <i
                                                className="fas fa-user-plus"
                                    />
                                    Sign Up
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="nav-link"
                                    exact={true}
                                    onClick={undefined}
                                    to="/login"
                        >
                                    <i
                                                className="fas fa-sign-in-alt"
                                    />
                                    Login
                        </NavLink>
            </div>
</nav>,
        ],
        "className": "navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-film"
/>,
              "MovieTracker",
            ],
            "className": "nav-link brand",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-film",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "MovieTracker",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav-link"
                            exact={true}
                            onClick={undefined}
                            to="/register"
              >
                            <i
                                          className="fas fa-user-plus"
                            />
                            Sign Up
              </NavLink>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav-link"
                            exact={true}
                            onClick={undefined}
                            to="/login"
              >
                            <i
                                          className="fas fa-sign-in-alt"
                            />
                            Login
              </NavLink>
</div>,
            "className": "nav-btns",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  exact={true}
                  onClick={undefined}
                  to="/register"
>
                  <i
                                    className="fas fa-user-plus"
                  />
                  Sign Up
</NavLink>,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  exact={true}
                  onClick={undefined}
                  to="/login"
>
                  <i
                                    className="fas fa-sign-in-alt"
                  />
                  Login
</NavLink>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": Array [
                    <i
                      className="fas fa-user-plus"
/>,
                    "Sign Up",
                  ],
                  "className": "nav-link",
                  "exact": true,
                  "onClick": undefined,
                  "to": "/register",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-user-plus",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "Sign Up",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": Array [
                    <i
                      className="fas fa-sign-in-alt"
/>,
                    "Login",
                  ],
                  "className": "nav-link",
                  "exact": true,
                  "onClick": undefined,
                  "to": "/login",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-sign-in-alt",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "Login",
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "nav",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Navigation should match snapshot when currentUser exists 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    clearFavorites={[Function]}
    currentUser={
        Object {
            "email": "asdf",
            "id": 2,
            "name": "asdf",
            "password": "asdf",
          }
    }
    setCurrentUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="nav-link brand"
          replace={false}
          to="/"
>
          <i
                    className="fas fa-film"
          />
          MovieTracker
</Link>,
        <nav
          className="nav-btns"
>
          <div>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav-link"
                              exact={true}
                              to="/favorites"
                    >
                              <i
                                        className="fas fa-heart"
                              />
                              Favorites
                    </NavLink>
                    <Link
                              className="nav-link nav-link-logout"
                              exact={true}
                              onClick={[Function]}
                              replace={false}
                              to="/"
                    >
                              Logout
                    </Link>
          </div>
</nav>,
      ],
      "className": "navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-film"
/>,
            "MovieTracker",
          ],
          "className": "nav-link brand",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-film",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "MovieTracker",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        to="/favorites"
            >
                        <i
                                    className="fas fa-heart"
                        />
                        Favorites
            </NavLink>
            <Link
                        className="nav-link nav-link-logout"
                        exact={true}
                        onClick={[Function]}
                        replace={false}
                        to="/"
            >
                        Logout
            </Link>
</div>,
          "className": "nav-btns",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                exact={true}
                to="/favorites"
>
                <i
                                className="fas fa-heart"
                />
                Favorites
</NavLink>,
              <Link
                className="nav-link nav-link-logout"
                exact={true}
                onClick={[Function]}
                replace={false}
                to="/"
>
                Logout
</Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": Array [
                  <i
                    className="fas fa-heart"
/>,
                  "Favorites",
                ],
                "className": "nav-link",
                "exact": true,
                "to": "/favorites",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-heart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Favorites",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Logout",
                "className": "nav-link nav-link-logout",
                "exact": true,
                "onClick": [Function],
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Logout",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="nav-link brand"
            replace={false}
            to="/"
>
            <i
                        className="fas fa-film"
            />
            MovieTracker
</Link>,
          <nav
            className="nav-btns"
>
            <div>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="nav-link"
                                    exact={true}
                                    to="/favorites"
                        >
                                    <i
                                                className="fas fa-heart"
                                    />
                                    Favorites
                        </NavLink>
                        <Link
                                    className="nav-link nav-link-logout"
                                    exact={true}
                                    onClick={[Function]}
                                    replace={false}
                                    to="/"
                        >
                                    Logout
                        </Link>
            </div>
</nav>,
        ],
        "className": "navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-film"
/>,
              "MovieTracker",
            ],
            "className": "nav-link brand",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-film",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "MovieTracker",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav-link"
                            exact={true}
                            to="/favorites"
              >
                            <i
                                          className="fas fa-heart"
                            />
                            Favorites
              </NavLink>
              <Link
                            className="nav-link nav-link-logout"
                            exact={true}
                            onClick={[Function]}
                            replace={false}
                            to="/"
              >
                            Logout
              </Link>
</div>,
            "className": "nav-btns",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  exact={true}
                  to="/favorites"
>
                  <i
                                    className="fas fa-heart"
                  />
                  Favorites
</NavLink>,
                <Link
                  className="nav-link nav-link-logout"
                  exact={true}
                  onClick={[Function]}
                  replace={false}
                  to="/"
>
                  Logout
</Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": Array [
                    <i
                      className="fas fa-heart"
/>,
                    "Favorites",
                  ],
                  "className": "nav-link",
                  "exact": true,
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-heart",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "Favorites",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Logout",
                  "className": "nav-link nav-link-logout",
                  "exact": true,
                  "onClick": [Function],
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Logout",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "nav",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
